fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStoreTest.java

```diff
@@ -276,4 +280,135 @@ public void withToProto() throws Exception {
+                            Index.EMPTY_VALUE, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+            metaDataStore.addMultiTypeIndex(null,
+                    new Index("all$elements2", Key.Expressions.field("element", KeyExpression.FanType.Concatenate),
+                            Index.EMPTY_VALUE, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+            metaDataStore.addMultiTypeIndex(Arrays.asList("MultiRecordTwo", "MultiRecordThree"),
+                    new Index("two&three$ego", Key.Expressions.field("ego"), Index.EMPTY_VALUE, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+            metaDataStore.addMultiTypeIndex(Arrays.asList("MultiRecordOne"), new Index("one$name", Key.Expressions.field("name"), IndexTypes.VALUE));
+            context.commit();
+        }
+
+        try (FDBRecordContext context = fdb.openContext()) {
+            openMetaDataStore(context);
+            RecordMetaData metaData = metaDataStore.getRecordMetaData();
+            assertEquals(5, metaData.getAllIndexes().size());
+            assertEquals(0, metaData.getFormerIndexes().size());
+            assertNotNull(metaData.getIndex("all$elements"));
+            assertEquals(3, metaData.recordTypesForIndex(metaData.getIndex("all$elements")).size());
+            assertNotNull(metaData.getIndex("all$elements2"));
+            assertEquals(3, metaData.recordTypesForIndex(metaData.getIndex("all$elements2")).size());
+            assertNotNull(metaData.getIndex("two&three$ego"));
+            assertEquals(2, metaData.recordTypesForIndex(metaData.getIndex("two&three$ego")).size());
+            assertNotNull(metaData.getIndex("one$name"));
+            assertEquals(1, metaData.recordTypesForIndex(metaData.getIndex("one$name")).size());
+            metaDataStore.dropIndex("one$name");
+            MetaDataException e = assertThrows(MetaDataException.class, () ->
+                    metaDataStore.getRecordMetaData().getIndex("one$name"));
+            assertEquals("Index one$name not defined", e.getMessage());
+            context.commit();
+        }
+
+        try (FDBRecordContext context = fdb.openContext()) {
+            openMetaDataStore(context);
+            RecordMetaData metaData = metaDataStore.getRecordMetaData();
+            assertEquals(4, metaData.getAllIndexes().size());
+            assertEquals(1, metaData.getFormerIndexes().size());
+            assertNotNull(metaDataStore.getRecordMetaData().getIndex("all$elements"));
+            assertNotNull(metaDataStore.getRecordMetaData().getIndex("all$elements2"));
+            assertNotNull(metaDataStore.getRecordMetaData().getIndex("two&three$ego"));
+            MetaDataException e = assertThrows(MetaDataException.class, () ->
+                    metaDataStore.getRecordMetaData().getIndex("one$name"));
+            assertEquals("Index one$name not defined", e.getMessage());
+        }
+    }
+
+    @Test
+    public void updateRecords() {
```

Comment:
I think a test that combines `updateRecords` with a local file descriptor would be nice.

Reply:
Where?